graph/appender/dead_node.go

```diff
@@ -21,10 +27,52 @@ func (a DeadNodeAppender) AppendGraph(trafficMap graph.TrafficMap, _ string) {
 	business, err := business.Get()
 	checkError(err)
 
-	applyDeadNodes(trafficMap, business.Workload)
+	if a.serviceEntries == nil {
+		a.serviceEntries = fetchServiceEntries(business)
+	}
+
+	applyDeadNodes(trafficMap, business.Workload, a.serviceEntries)
+}
+
+// fetchServiceEntries queries the cluster API to retrieve ServiceEntries resources
+// across all namespaces in the cluster.
+// All ServiceEntries are needed, because Istio does not distinguish where the
+// ServiceEntries are created when routing egress traffic.
+func fetchServiceEntries(bLayer *business.Layer) models.ServiceEntries {
+	namespaces, err := bLayer.Namespace.GetNamespaces()
+	checkError(err)
+
+	serviceEntries := models.ServiceEntries{}
+	for _, ns := range namespaces {
+		istioCfg, err := bLayer.IstioConfig.GetIstioConfig(business.IstioConfigCriteria{
+			IncludeServiceEntries: true,
+			Namespace:             ns.Name,
+		})
+		checkError(err)
+
+		serviceEntries = append(serviceEntries, istioCfg.ServiceEntries...)
+	}
+
+	return serviceEntries
+}
+
+// isExternalService checks whether node represents an external service (egress traffic), looking
+// for a matching egress rule in serviceEntries.
+func isExternalService(node *graph.Node, serviceEntries models.ServiceEntries) bool {
```

Comment:
Should this be passing by reference? (e.g. use "serviceEntries *models.ServiceEntries") --- this would avoid having to make a copy these entries every time we check a node. I'm not sure now large these ServiceEntries maps are, maybe its not a big deal. But it seems if we are going to call this for every node in the graph, we should think about passing this cache of service entries by reference.

Reply:
Makes sense! Asterisk added.
Probably ServiceEntries may not be large... But who knows?... Only users I guess.

Reply:
Actually slices already hold a pointer to the underlying array, so I think it wasn't necessary.

Reply:
@jotak I though that was the case. But added the reference just in case... It doesn't hurt, so ... I'll keep the asterisk :)